core/src/main/scala/akka/persistence/jdbc/util/PluginVersionChecker.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.util
+
+object PluginVersionChecker {
+  def check() =
+    try {
+      Class.forName("akka.persistence.jdbc.util.DefaultSlickDatabaseProvider")
+      throw new RuntimeException(
+        "Old version of Akka Persistence JDBC found on the classpath. Remove `com.github.dnvriend:akka-persistence-jdbc` from the classpath..")
+    } catch {
+      case _: ClassNotFoundException =>
```

Comment:
In corresponding Akka utility we also handle `NoClassDefFoundError` but maybe that is not needed here?
https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/actor/ReflectiveDynamicAccess.scala#L51

Reply:
If I understand correctly, `NoClassDefFoundError` happens when the bytecode references it and it's not found at runtime. So, not applicable here. 

Btw, I didn't think about it. I just got inspiration (copied) from the akka discovery check. There we only catch `ClassNotFoundException`. And probably for the same reasons, we don't depend on it at bytecode level.